tests/atest/conftest.py

```diff
@@ -78,4 +78,9 @@ def pytest_generate_tests(metafunc):
         ['-c', 'section-out-of-order:sections_order:settings,variables,testcases,tasks,keywords'],
         'duplications/section-out-of-order_default_order'
     ))
+    auto_discovered_rules.append((
+        'tag-with-reserved',
+        [],
+        'tags/tag-with-reserved-in-keyword-doc'
+    ))
```

Comment:
Special test data (for tags in documentation_ was added but not executed because the directory name is different from rule name.